subprojects/core-api/src/main/java/org/gradle/api/Project.java

```diff
@@ -237,6 +237,14 @@ public interface Project extends Comparable<Project>, ExtensionAware, PluginAwar
 
     String GRADLE_PROPERTIES = "gradle.properties";
 
+    /**
+     * The properties file to store build configuration.
+     *
+     * @since 8.7
+     */
+    @Incubating
+    String BUILD_PROPERTIES = "gradle/gradle-build.properties";
```

Comment:
Not exactly necessary to expose this public field under `Project`, however, makes completely sense to have it declared here since this file keeps references to directories about `build` and `properties` like `gradle.properties`, keeping consistency. Not sure if there will be another better place of declaring this without making visible to the public API

Reply:
Like before, we should have a strong use case for adding public APIs and I do not see that here. 

When do we expect users to use this field? They should not be writing code to modify this file -- that is the job of the update task. 

And, even if we did identify a use case for this field, it would not belong on `Project`. 

Yes, there already exists a similar field defined on `Project`, but that was added back in 2008 https://github.com/gradle/gradle/commit/5dd2466a43081404cf93f67313aadb118f8bb050#diff-1160567c5abb4bcabb66ca1363c30a45c3b4ce0e57b8c96b41c6a3ca21934314

This field has nothing to do with a project